test/test_torch.py

```diff
@@ -18073,6 +18073,30 @@ def compare_with_numpy(contiguous_input=True):
         self.assertEqual(a[5:].real, a.real[5:])
         self.assertEqual(a[5:].imag, a.imag[5:])
 
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_complex_dtypes())
+    def test_set_real_imag(self, device, dtype):
+        x = torch.randn(10, dtype=dtype)
+
+        for d in torch.testing.get_all_dtypes():
+            new_real = _make_tensor((10,), d, device)
+            new_imag = _make_tensor((10,), d, device)
+
+            if d.is_complex:
+                regex = "Casting complex values to real discards the imaginary part"
+                with self.assertWarnsRegex(UserWarning, regex):
+                    x.real = new_real
+                with self.assertWarnsRegex(UserWarning, regex):
+                    x.imag = new_imag
+                self.assertEqualIgnoreType(x.real, new_real.real)
+                self.assertEqualIgnoreType(x.imag, new_imag.real)
+
+            else:
+                x.real = new_real
+                x.imag = new_imag
+                self.assertEqualIgnoreType(x.real, new_real)
+                self.assertEqualIgnoreType(x.imag, new_imag)
```

Comment:
Same here. assertEqualIgnoreType will eventually be removed, so we don't want to increase uses of it.